stellar-contract-env-host/src/host.rs

```diff
@@ -1549,30 +1549,131 @@ impl CheckedEnv for Host {
         Ok(self.to_host_obj(&sco)?.into())
     }
 
-    fn binary_copy_to_guest_mem(
+    fn binary_copy_to_linear_memory(
         &self,
         b: Object,
         i: RawVal,
         j: RawVal,
         l: RawVal,
     ) -> Result<RawVal, HostError> {
-        todo!()
+        //[offset in the host binary, pos in the linear memory, length of the slice to copy]
```

Comment:
We could name the parameters with these names instead of `i`, `j`, `l`, then the comment would be less necessary.

Reply:
Thanks, fixed.

Reply:
I was meaning can we name the parameters in the function itself, all the way up in the host fn definition in the macro? It looks like ijl populate in a few places, including into the SDK PR, which isn't ideal.

Reply:
I see. Are you just referring to these functions, or host function parameter naming in general? Since all current hostfn parameters are 1 char (at most 2) in length and it helps keeping things concise and compact (not sure if there is impact on code size too). I'm not sure if it's worth changing these names to break from the convention. Although I do see these indices are a bit less obvious than the other hostfns.
It would be look like:
`{"2", fn binary_copy_from_linear_memory(b:Object, offset_hv:RawVal, pos_lm:RawVal, len:RawVal) -> Object}`

Reply:
>Are you just referring to these functions, or host function parameter naming in general? 

Most other functions are easily understood even if the arguments are single letters. For example, the following is reasonably understood that `a` will be an account and `s` will be a signer.
https://github.com/stellar/rs-stellar-contract-env/blob/e9128da4635e23053084e24359d42f9348690bad/stellar-contract-env-common/src/env.rs#L302

But some functions it is very unclear. The letters `i`, `j`, and `l` don't appear to align at all semantically with what is intended to be passed in to the function. I also think it is unlikely that if we used names like `o`, `p`, and `l`, that it would help. Some host functions need more descriptive names. We've used more descriptive names for some functions before, so I don't think we're breaking from convention. For example:
https://github.com/stellar/rs-stellar-contract-env/blob/e9128da4635e23053084e24359d42f9348690bad/stellar-contract-env-common/src/env.rs#L159-L160
https://github.com/stellar/rs-stellar-contract-env/blob/e9128da4635e23053084e24359d42f9348690bad/stellar-contract-env-common/src/env.rs#L166

Reply:
Might I suggest something like:

```
{"2", fn binary_copy_from_linear_memory(b:Object, b_pos:RawVal, lm_pos:RawVal, len:RawVal) -> Object}
```

Reply:
> Some host functions need more descriptive names. We've used more descriptive names for some functions before, so I don't think we're breaking from convention. For example:

Ah yes, didn't see those. 

Thank you, I will update those names.